mne/io/fieldtrip/utils.py

```diff
@@ -0,0 +1,322 @@
+
+    events = np.vstack([np.array(event_number), event_trans_val,
+                        event_type]).astype('int').T
+
+    return events
+
+
+def _create_event_metadata(ft_struct):
+    """Create event metadata from trialinfo."""
+    pandas = _check_pandas_installed(strict=False)
+    if not pandas:
+        warn('The Pandas library is not installed. Not returning the original '
+             'trialinfo matrix as metadata.')
+        return None
+
+    metadata = pandas.DataFrame(ft_struct['trialinfo'])
+
+    return metadata
+
+
+def _process_channel_eeg(cur_ch, elec):
+    """Convert EEG channel from FieldTrip to MNE.
+
+    Parameters
+    ----------
+    cur_ch: dict
+        Channel specific dictionary to populate.
+
+    elec: dict
+        elec dict as loaded from the FieldTrip structure
+
+    Returns
+    -------
+    dict: The original dict (cur_ch) with the added information
+    """
+    all_labels = np.asanyarray(elec['label'])
+    chan_idx_in_elec = np.where(all_labels == cur_ch['ch_name'])[0][0]
+    position = np.squeeze(elec['chanpos'][chan_idx_in_elec, :])
+    chanunit = elec['chanunit'][chan_idx_in_elec]
+    position_unit = elec['unit']
+
+    position = position * _unit_dict[position_unit]
+    cur_ch['loc'] = np.hstack((position, np.zeros((9,))))
+    cur_ch['loc'][-1] = 1
+    cur_ch['unit'] = FIFF.FIFF_UNIT_V
+    cur_ch['unit_mul'] = np.log10(_unit_dict[chanunit[0]])
+    cur_ch['kind'] = FIFF.FIFFV_EEG_CH
+    cur_ch['coil_type'] = FIFF.FIFFV_COIL_EEG
+    cur_ch['coord_frame'] = FIFF.FIFFV_COORD_HEAD
+
+    return cur_ch
+
+
+def _process_channel_meg(cur_ch, grad):
+    """Convert MEG channel from FieldTrip to MNE.
+
+    Parameters
+    ----------
+    cur_ch: dict
+        Channel specific dictionary to populate.
+
+    grad: dict
+        grad dict as loaded from the FieldTrip structure
+
+    Returns
+    -------
+    dict: The original dict (cur_ch) with the added information
+    """
+    all_labels = np.asanyarray(grad['label'])
+    chan_idx_in_grad = np.where(all_labels == cur_ch['ch_name'])[0][0]
+    gradtype = grad['type']
+    chantype = grad['chantype'][chan_idx_in_grad]
+    position_unit = grad['unit']
+    position = np.squeeze(grad['chanpos'][chan_idx_in_grad, :])
+    position = position * _unit_dict[position_unit]
+
+    if gradtype == 'neuromag306' and 'tra' in grad and 'coilpos' in grad:
+        # Try to regenerate original channel pos.
+        idx_in_coilpos = np.where(grad['tra'][chan_idx_in_grad, :] != 0)[0]
+        cur_coilpos = grad['coilpos'][idx_in_coilpos, :]
+        cur_coilpos = cur_coilpos * _unit_dict[position_unit]
+        cur_coilori = grad['coilori'][idx_in_coilpos, :]
+        if chantype == 'megmag':
+            position = cur_coilpos[0] - 0.0003 * cur_coilori[0]
+        if chantype == 'megplanar':
+            tmp_pos = cur_coilpos - 0.0003 * cur_coilori
+            position = np.average(tmp_pos, axis=0)
+
+    # TODO: this needs to be fixed!
```

Comment:
?

Reply:
That was a leftover from times when we still wanted to get proper sensor orientations from the fieldtrip files.